tests/test_similarityrecommender.py

```diff
@@ -0,0 +1,194 @@
+from moto import mock_s3
+import boto3
+import json
+import numpy as np
+import scipy
+from taar.recommenders.similarity_recommender import SimilarityRecommender
+
+FAKE_DONOR_DATA = [
+    {"activeAddons":
+        ["{test-guid-1}", "{test-guid-2}", "{test-guid-3}", "{test-guid-4}"],
+     "geoCity": "nowhere-us",
+     "subsessionLength": 6500,
+     "locale": "en-US",
+     "os": "mac",
+     "bookmark_count": 77,
+     "tab_open_count": 32,
+     "total_uri": 87,
+     "unique_tlds": 65
+     },
+    {"activeAddons":
+        ["{test-guid-5}", "{test-guid-6}", "{test-guid-7}", "{test-guid-8}"],
+     "geoCity": "pompei-it",
+     "subsessionLength": 67832,
+     "locale": "it-IT",
+     "os": "Linux",
+     "bookmark_count": 8166,
+     "tab_open_count": 232,
+     "total_uri": 42342,
+     "unique_tlds": 1203
+     },
+    {"activeAddons":
+        ["{test-guid-9}", "{test-guid-10}", "{test-guid-11}", "{test-guid-12}"],
+     "geoCity": "brasilia-br",
+     "subsessionLength": 1011,
+     "locale": "br-PT",
+     "os": "windows",
+     "bookmark_count": 17,
+     "tab_open_count": 3,
+     "total_uri": 432,
+     "unique_tlds": 10
+     }
+]
+
+
+def generate_fake_lr_curves(num_elements):
+    lr_index = list(np.linspace(0, 10, num_elements))
+    numerator_density = [scipy.stats.norm.pdf(float(i), 0.5, 0.5) for i in lr_index]
+    denominator_density = [scipy.stats.norm.pdf(float(i), 5.0, 1.5) for i in lr_index]
+
+    # JSON conversion breaks the tuple, so lists are needed.
+    p_pairs = [list(a) for a in zip(numerator_density, denominator_density)]
```

Comment:
I think this line and the one above can simply be summarized as:
`return zip(lr_index, zip(numerator_density, denominator_density))`

Reply:
That would return a list of tuples, which causes problems in testing becasue the JSON export breaks them. the conversion to list of list emulating tuples fixes that wierdness.

Reply:
The output from this function, as you mentioned, is passed to `json.dumps()`. The output of that function will be identical regardless of using `zip` or not. You can quickly convince yourself about this with the following code, that you can put right below `p_pairs` or write in a python console:

```
original =  [list(a) for a in zip(lr_index, p_pairs)]
zip_version = zip(lr_index, zip(numerator_density, denominator_density))
assert json.dumps(original) == json.dumps(zip_version)
```

It won't throw, so the output is indeed equal :)

Reply:
yup, you are right, don't know why i had wierd behaviour before. Thanks for th catch.